apps/src/applab/applab.js

```diff
@@ -262,6 +262,67 @@ function renderFooterInSharedGame() {
   );
 }
 
+Applab.autogenerateML = function(modelId) {
+  $.ajax({
+    url: `/api/v1/ml_models/${modelId}`,
+    method: 'GET'
+  }).then(modelData => {
+    var x = 20;
+    var y = 20;
+    var SPACE = 20;
+    designMode.onInsertEvent(`var testValues = {};`);
+    modelData.selectedFeatures.forEach(feature => {
+      y = y + SPACE;
+      var label = designMode.createElement('LABEL', x, y);
+      label.textContent = feature + ':';
+      label.id = 'design_' + feature + '_label';
+      label.style.width = '300px';
+      y = y + SPACE;
+      if (Object.keys(modelData.featureNumberKey).includes(feature)) {
+        var selectId = feature + '_dropdown';
+        var select = designMode.createElement('DROPDOWN', x, y);
+        select.id = 'design_' + selectId;
+        // App Lab automatically addss "option 1" and "option 2", remove them.
+        select.options.remove(0);
+        select.options.remove(0);
+        Object.keys(modelData.featureNumberKey[feature]).forEach(option => {
+          var optionElement = document.createElement('option');
+          optionElement.text = option;
+          select.options.add(optionElement);
+        });
+      } else {
+        var input = designMode.createElement('TEXT_INPUT');
+        input.id = 'design_' + feature + '_input';
+      }
+      var addFeature = `testValues.${feature} = getText("${selectId}");`;
+      designMode.onInsertEvent(addFeature);
+    });
+    y = y + 2 * SPACE;
+    var label = designMode.createElement('LABEL', x, y);
+    label.textContent = modelData.labelColumn;
+    // TODO: this could be problematic if the name isn't formatted appropriately
```

Comment:
Names with whitespace cause this warning in App Lab when the blocks are generated. `WARNING: Line: 2: The onEvent() id parameter refers to an id ("did this save?_predict") which contains whitespace. Change the id name to ("didthissave?_predict")`
We could either 1.) strip whitespace when we use the name to generate ids for UI elements OR 2.) strip whitespace from names on save from AI Lab.  I'm leaning toward option 1.   Still need to see what happens with special characters, 😄 etc.

Reply:
These are the column names?  We could also remove spaces way back at import time into AI Lab?  And our precanned datasets could just be appropriately named already?  It would probably make all of our code safer, since I think AI Lab uses these column names for object keys too..

Reply:
These are model names that the user inputs. Then I was using the model name to name UI elements, like on the line below. I guess we could 3.) use model id instead e.g.  `label.id = 'design_' + modelData.id + '_label';`